internal/controller/pkg/revision/imageback.go

```diff
@@ -85,45 +88,70 @@ func (i *ImageBackend) Init(ctx context.Context, bo ...parser.BackendOption) (io
 	for _, o := range bo {
 		o(n)
 	}
-	var img regv1.Image
-	var err error
-
-	pullPolicy := n.pr.GetPackagePullPolicy()
-	if pullPolicy != nil && *pullPolicy == corev1.PullNever {
-		// If package is pre-cached we assume there are never multiple tags in
-		// the same image.
-		img, err = i.cache.Get("", n.pr.GetSource())
-		if err != nil {
-			return nil, errors.Wrap(err, errPullPolicyNever)
+	ref, err := name.ParseReference(n.pr.GetSource(), name.WithDefaultRegistry(i.registry))
+	if err != nil {
+		return nil, errors.Wrap(err, errBadReference)
+	}
+	// Fetch image from registry.
+	img, err := i.fetcher.Fetch(ctx, ref, v1.RefNames(n.pr.GetPackagePullSecrets())...)
+	if err != nil {
+		return nil, errors.Wrap(err, errFetchPackage)
+	}
+	// Get image manifest.
+	manifest, err := img.Manifest()
+	if err != nil {
+		return nil, errors.Wrap(err, errGetManifest)
+	}
+	// Determine if the image is using annotated layers.
+	var tarc io.ReadCloser
+	foundAnnotated := false
+	for _, l := range manifest.Layers {
+		if a, ok := l.Annotations[layerAnnotation]; !ok || a != baseAnnotationValue {
+			continue
+		}
+		// NOTE(hasheddan): the xpkg specification dictates that only one layer
+		// descriptor may be annotated as xpkg base. Since iterating through all
+		// descriptors is relatively inexpensive, we opt to do so in order to
+		// verify that we aren't just using the first layer annotated as xpkg
+		// base.
+		if foundAnnotated {
+			return nil, errors.New(errMultipleAnnotatedLayers)
```

Comment:
Would it make sense to fall back to using the artifacts from the 'flattened' image in this case? I'm thinking no; it would perhaps be more compatible, but also less predictable and in violation of the spec.